src/dataMap.js

```diff
@@ -15,852 +15,929 @@ import { extendArray, to2dArray } from './helpers/array';
-    const schemaLen = deepObjectSize(schema);
+      if (typeof columns === 'function') {
+        columnsLen = schemaLen > 0 ? schemaLen : this.instance.countSourceCols();
+        columnsAsFunc = true;
+      }
 
-    if (typeof columns === 'function') {
-      columnsLen = schemaLen > 0 ? schemaLen : this.instance.countSourceCols();
-      columnsAsFunc = true;
-    }
+      for (i = 0; i < columnsLen; i++) {
+        const column = columnsAsFunc ? columns(i) : columns[i];
 
-    for (i = 0; i < columnsLen; i++) {
-      const column = columnsAsFunc ? columns(i) : columns[i];
+        if (isObject(column)) {
+          if (typeof column.data !== 'undefined') {
+            const index = columnsAsFunc ? filteredIndex : i;
+            this.colToPropCache[index] = column.data;
+            this.propToColCache.set(column.data, index);
+          }
 
-      if (isObject(column)) {
-        if (typeof column.data !== 'undefined') {
-          const index = columnsAsFunc ? filteredIndex : i;
-          this.colToPropCache[index] = column.data;
-          this.propToColCache.set(column.data, index);
+          filteredIndex += 1;
         }
-
-        filteredIndex += 1;
       }
-    }
 
-  } else {
-    this.recursiveDuckColumns(schema);
+    } else {
+      this.recursiveDuckColumns(schema);
+    }
   }
-};
 
-/**
- * Returns property name that corresponds with the given column index.
- *
- * @param {Number} col Visual column index.
- * @returns {Number} Physical column index.
- */
-DataMap.prototype.colToProp = function(col) {
-  const physicalColumn = this.instance.runHooks('modifyCol', col);
+  /**
+   * Generates columns' translation cache.
+   *
+   * @param {Object} schema
+   * @param {Number} lastCol
+   * @param {Number} parent
+   * @returns {Number}
+   */
+  recursiveDuckColumns(schema, lastCol, parent) {
+    let lastColumn = lastCol;
+    let propertyParent = parent;
+    let prop;
+
+    if (typeof lastColumn === 'undefined') {
+      lastColumn = 0;
+      propertyParent = '';
+    }
+    if (typeof schema === 'object' && !Array.isArray(schema)) {
+      objectEach(schema, (value, key) => {
+        if (value === null) {
+          prop = propertyParent + key;
+          this.colToPropCache.push(prop);
+          this.propToColCache.set(prop, lastColumn);
+
+          lastColumn += 1;
+        } else {
+          lastColumn = this.recursiveDuckColumns(value, lastColumn, `${key}.`);
+        }
+      });
+    }
 
-  if (!isNaN(physicalColumn) && this.colToPropCache && typeof this.colToPropCache[physicalColumn] !== 'undefined') {
-    return this.colToPropCache[physicalColumn];
+    return lastColumn;
   }
 
-  return physicalColumn;
-};
+  /**
+   * Returns property name that corresponds with the given column index.
+   *
+   * @param {Number} col Visual column index.
+   * @returns {Number} Physical column index.
+   */
+  colToProp(col) {
+    const physicalColumn = this.instance.runHooks('modifyCol', col);
```

Comment:
Can I ask you to rewrite this hook to a more descriptive method? Such as:
```suggestion
    const physicalColumn = this.instance.toPhysicalColumn(col);
```